bluepysnap/nodes.py

```diff
@@ -394,10 +599,15 @@ def ids(self, group=None, limit=None, sample=None):
                 Otherwise the result is sorted and contains no duplicates.
 
             sample (int): If specified, randomly choose ``sample`` number of
-                IDs from the match result.
+                IDs from the match result. If the size of the sample is greater than
+                the size of the NodePopulation then all ids are taken and shuffled.
 
             limit (int): If specified, return the first ``limit`` number of
-                IDs from the match result.
+                IDs from the match result. If limit is greater than the size of the population
+                all node IDs are returned.
+
+            raise_missing_property (bool): if True, raises if a property is not listed in this
+                population. Otherwise the ids are just not selected if a property is missing.
 
         Returns:
             numpy.array: A numpy array of IDs.
```

Comment:
You should probably add an example with circuitnodeids in the example list below

Reply:
done